src/main/java/com/jaehyun/reservation/user/user/service/UserService.java

```diff
@@ -63,4 +64,28 @@ public APIResponse<Void> quit(Principal principal) {
     userRepository.deleteById(user.get().getId());
     return APIResponse.delete();
   }
+
+  public APIResponse<String> changeRole(Principal principal) {
+    Optional<User> userOptional = userRepository.findByLoginId(principal.getName());
+
+    if (userOptional.isPresent()) {
+      User user = userOptional.get();
+
+      User updatedUser = User.builder()
+          .id(user.getId())
+          .loginId(user.getLoginId())
+          .phoneNum(user.getPhoneNum())
+          .password(user.getPassword())
+          .name(user.getName())
+          .roles(RoleType.ADMIN) // 변경할 역할 값으로 설정
+          .reviewList(user.getReviewList())
+          .storeList(user.getStoreList())
+          .favoriteList(user.getFavoriteList())
+          .build();
+      userRepository.saveAndFlush(updatedUser);
+      return APIResponse.success(API_NAME, user.getName());
+    } else {
+      throw new BadTokenException();
+    }
```

Comment:
orElseThrow를 사용하시면 좋을듯하구요, user의 role을 업데이트하는 메소드로 보이는데 entity를 새로 만드는 이유가 있나요? repository에서 가지고 온 유저에서 role만 바꿔주면 되지 않나 싶습니다

```suggestion.
    User user = userRepository.findByLoginId(principal.getName()).orElseThrow(() -> new BadTokenException());
    user.setRoles(RoleType.ADMIN);
    userRepository.saveAndFlush(user);
```